projects/mf-dynamic-form/src/lib/components/df-form-control/df-form-control.component.ts

```diff
@@ -354,29 +355,35 @@ export class DfFormControlComponent implements OnInit, OnDestroy {
     const promise: Promise<DropdownOption[]> = options$.callback(value);
 
     this.logPromiseFetch(value)
-    promise.then((value: DropdownOption[]) => {
-      this.control.options = value
-      this.form.controls[this.control.key].setValue(value.length ? value[0].value : null);
+    promise.then((option: DropdownOption[]) => {
+      this.control.options = option;
+      this.form.controls[this.control.key].setValue(this.getFieldValue(option));
       this.form.controls[this.control.key].updateValueAndValidity();
     })
 
-      this.subx.push(this.form.get(fieldName).valueChanges.subscribe(value => {
+    this.subx.push(this.form.get(fieldName).valueChanges.subscribe(value => {
 
-        const promise: Promise<DropdownOption[]> = options$.callback(value);
+      const promise: Promise<DropdownOption[]> = options$.callback(value);
 
-        this.logPromiseFetch(value)
-        promise.then((value: DropdownOption[]) => {
-          this.control.options = value
-          this.form.controls[this.control.key].setValue(value.length ? value[0].value : null);
-          this.form.controls[this.control.key].updateValueAndValidity();
-        })
+      this.logPromiseFetch(value)
+      promise.then((option: DropdownOption[]) => {
+        this.control.options = option;
+        this.form.controls[this.control.key].setValue(this.getFieldValue(option));
+        this.form.controls[this.control.key].updateValueAndValidity();
+      })
 
-      }));
+    }));
+  }
+
+  private getFieldValue(value: DropdownOption[]) {
+    const checkValueExist = value.some(it => it.value == this.control.value);
```

Comment:
`const selectedOption = options.find(it => it.value == this.control.value)?.value;
return selectedOption?.length ? selectedOption[0].value : null;`

Reply:
si this.control.value est null un exception affiche
const selectedOption = options.find(it => it.value == this.control.value)?.value; 

il faut teste avant d'utilise find

Reply:
t'as raison, j'ai corrigé la proposition